plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -130,34 +143,26 @@ public Optional<RemoteDatabaseObject> toRemoteDataset(String projectId, String d
             return Optional.of(RemoteDatabaseObject.of(datasetName));
         }
 
-        Map<String, Optional<RemoteDatabaseObject>> mapping = new HashMap<>();
-        for (DatasetId datasetId : listDatasetIds(projectId)) {
-            mapping.merge(
-                    datasetId.getDataset().toLowerCase(ENGLISH),
-                    Optional.of(RemoteDatabaseObject.of(datasetId.getDataset())),
-                    (currentValue, collision) -> currentValue.map(current -> current.registerCollision(collision.get().getOnlyRemoteName())));
+        try {
+            Map<String, RemoteDatabaseObject> datasetMap = remoteDatasetCaseInsentiveCache.get(projectId, () -> {
+                Map<String, RemoteDatabaseObject> mapping = new HashMap<>();
+                for (DatasetId dataset : listDatasetIds(projectId)) {
+                    mapping.merge(
+                            dataset.getDataset().toLowerCase(ENGLISH),
```

Comment:
nit: introduce `datasetToLowerCase()`

Reply:
`DatasetId` exists in Google library. We can't add the new method.